daemon_windows.go

```diff
@@ -0,0 +1,29 @@
+package godaemon
+
+// Copyright (c) 2013-2015 VividCortex, Inc. All rights reserved.
+// Please see the LICENSE file for applicable license terms.
+
+import (
+	"fmt"
+	"syscall"
+	"unicode/utf16"
+	"unsafe"
+)
+
+var (
+	getModuleFileName = syscall.MustLoadDLL("kernel32.dll").MustFindProc("GetModuleFileNameW")
+)
+
+// GetExecutablePath returns the absolute path to the currently running
+// executable.  It is used internally by the godaemon package, and exported
+// publicly because it's useful outside of the package too.
+func GetExecutablePath() (string, error) {
+	buf := make([]uint16, syscall.MAX_PATH+1)
+
+	res, _, err := getModuleFileName.Call(0, uintptr(unsafe.Pointer(&buf[0])), uintptr(len(buf)))
+	if res == 0 || res >= syscall.MAX_PATH || buf[0] == 0 || buf[res-1] == 0 {
```

Comment:
I don't understand the two last conditions. Zero is not a valid character in a Windows file name, so a zero in the buffer is terminating the string, that would therefore have length zero, so `res` would be zero. (And that's a failure condition for the function.) Also, if the path could fit in the buffer `res` will be its length in characters, not including the terminating zero byte. So `buf[res-1]` would be non-zero by definition.